app/controllers/search_controller.rb

```diff
@@ -9,109 +9,48 @@ def index
 
     results_per_page = RESULTS_PER_PAGE
 
-    search_type = params.delete('search_type')
-
-    # Did the search come from quick_find or full_search
-    if search_type == 'quick_find'
-      # If search came from search, only use the search input
-      @query = {}
-
-      @query['keyword'] = params['keyword'] || ''
-      @query['record_state'] = 'published_records'
-      @query['sort_key'] = params['sort_key'] if params['sort_key']
-    elsif search_type == 'full_search'
-      # If search came from full search, ignore whatever was in search
-      @query = params.clone
-      @query['keyword'] ||= ''
-      @query.delete('provider_id') if @query['provider_id'].blank?
-    end
+    @record_type = params['record_type']
 
+    # set query
+    @query = {}
+    @query['keyword'] = params['keyword'] || ''
+    @query['provider_id'] = params['provider_id'] unless params['provider_id'].blank?
+    @query['sort_key'] = params['sort_key'] if params['sort_key']
```

Comment:
`provider_id` and `sort_key` seem like the same check but they are done with different code. Would increase readability if they check for these was the same, if possible.